### PR TITLE
Update contributing docs

### PR DESCRIPTION
**Pull request by neverett**
## Why are the changes needed?

Make contributing to code and docs easier for Flyte community members.

**Note:** merge this PR first, then merge https://github.com/flyteorg/flytesnacks/pull/1749, which depends on the changes in this PR.

## What changes were proposed in this pull request?

* Applied changes from https://github.com/flyteorg/flytesnacks/pull/1736 to the docs contributing guide in this repo, since the guide has moved
* Created a new "contributing" subsection, with separate pages on contributing code, docs, and flytesnacks examples (i.e. tutorials and integrations examples). I mostly left the content alone, just reorganized it a bit.

## How was this patch tested?

Local docs build.

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs

https://github.com/flyteorg/flytesnacks/pull/1736
https://github.com/flyteorg/flytesnacks/pull/1749

## Docs link

https://flyte--5830.org.readthedocs.build/en/5830/community/contribute/index.html

### CODE REVIEW

File: docs/community/index.rst [hunk 1]
```diff
@@ -30,7 +30,7 @@ Please join us on:
 Open Source Community Meeting 
 -----------------------------
 
-When: every other Tuesday, 9:00 AM Pacific Time.
+When: Every other Thursday, 11:00 AM Pacific Time.
```
Comment (neverett) [inline docs/community/index.rst:33]: @davidmirror-ops is this accurate? And is the event link below (https://www.addevent.com/event/EA7823958) still correct?

File: docs/community/contribute/contribute_docs.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+For minor edits that don't require a local setup, you can edit the page in GitHub page to propose improvements.
+
+## Contributing to user guide and deployment documentation
+
+To update user guide or deployment documentation, edit the corresponding files in the [flyte repository](https://github.com/flyteorg/flyte/tree/master/docs/user_guide).
+
+### Code in user guide documentation
+
+If you want to include tested, runnable example code in user guide documentation, you will need to add your code to the examples directory of the [flytesnacks repository](https://github.com/flyteorg/flytesnacks). Write your code in regular Python, with regular comments. These comments **will not** be extracted from the Python file and turned into user-facing documentation. You can use the `rli` ([remoteliteralinclude](https://github.com/wpilibsuite/sphinxext-remoteliteralinclude/blob/main/README.md)) directive to include snippets of code from your example Python file.
```
Comment (ppiegaze) [inline docs/community/contribute/contribute_docs.md:28]: This sentence reads oddly: "These comments **will not** be extracted from the Python file and turned into use-facing documentation"

The reader may be left wondering why exactly this particular fact about the comments is being called out.

**Review by ppiegaze — APPROVED** [review]
Minor comment inline, but generally looks good

File: docs/community/contribute/contribute_docs.md (hunk 2 above)
Comment (neverett) [reply docs/community/contribute/contribute_docs.md:28]: Good point -- this is to differentiate user guide examples in flytesnacks from tutorials and integrations examples in that repo, which have their comments turned into rendered docs. I can make that more explicit.

**Review by ppiegaze — APPROVED** [review]

**Review by ppiegaze — APPROVED** [review]

**Review by ppiegaze — APPROVED** [review]

**Pull request merged**